src/com/farmlogs/conduit/subscription.clj

```diff
@@ -34,49 +32,45 @@
                (a/>!! result-chan :drop)))))))
 
 (defn consume-ok
-  [{:keys [queue-name] :as config} consumer-tag]
+  [queue-name consumer-tag]
   (log/infof "Consuming %s with tag %s" queue-name consumer-tag))
 
 (defn cancel
-  [cancelled-promise {:keys [queue-name] :as queue-config} consumer-tag]
+  [cancelled-promise queue-name consumer-tag]
   (deliver cancelled-promise true)
   (log/warnf "Cancelled consumer tag: %s on queue: %s" consumer-tag queue-name))
 
 (defn cancel-ok
-  [cancelled-promise {:keys [queue-name] :as queue-config} consumer-tag]
+  [cancelled-promise queue-name consumer-tag]
   (deliver cancelled-promise true)
   (log/infof "Consumer tag: %s on queue: %s shutdown" consumer-tag queue-name))
 
 (defn- make-channel
-  [conn prefetch-count {:keys [exchange-name exchange-type] :as config}]
-  (let [chan (rmq.chan/open conn)]
-    (rmq.basic/qos chan prefetch-count)
-    (rmq.exch/declare chan exchange-name exchange-type config)
-    (rmq.queue/declare chan (:queue-name config) config)
-    (rmq.queue/bind chan (:queue-name config) exchange-name config)
-    chan))
+  [conn prefetch-count]
+  (doto (rmq.chan/open conn)
+    (rmq.basic/qos prefetch-count)))
```

Comment:
I don't think we've explained this thoroughly enough yet, we need to put together better documentation - this is how to use the plugin, but which repository do I implement the transformation in?

Once it's clear how people should be creating a worker that uses a new rabbitmq resource I'll be good for a +1.